stl/inc/mutex

```diff
@@ -508,61 +508,70 @@ public:
 };
 #endif // _HAS_CXX17
 
-// FUNCTION TEMPLATE _Invoke_stored_explicit
-template <class... _Types, size_t... _Indices>
-auto _Invoke_stored_explicit(tuple<_Types...>&& _Tuple, index_sequence<_Indices...>) -> decltype(
-    _STD invoke(_STD get<_Indices>(_STD move(_Tuple))...)) { // invoke() a tuple with explicit parameter ordering
-    return _STD invoke(_STD get<_Indices>(_STD move(_Tuple))...);
-}
-
-// FUNCTION TEMPLATE _Invoke_stored
-template <class... _Types>
-auto _Invoke_stored(tuple<_Types...>&& _Tuple)
-    -> decltype(_Invoke_stored_explicit(_STD move(_Tuple), index_sequence_for<_Types...>{})) { // invoke() a tuple
-    return _Invoke_stored_explicit(_STD move(_Tuple), index_sequence_for<_Types...>{});
-}
-
 // FUNCTION TEMPLATE call_once
-[[noreturn]] _CRTIMP2_PURE void __CLRCALL_PURE_OR_CDECL _XGetLastError();
-
-template <class _Tuple, class _Seq, size_t _Idx>
-int __stdcall _Callback_once(void*, void* _Pv, void**) { // adapt call_once() to callback API
-    _Tuple* _Ptup = static_cast<_Tuple*>(_Pv);
-
-    _TRY_BEGIN
-    // Note explicit _Seq{} selects every element from *_Ptup except the last,
-    // which contains call_once's exception_ptr.
-    _Invoke_stored_explicit(_STD move(*_Ptup), _Seq{});
-    _CATCH_ALL
-    auto& _Ref = _STD get<_Idx>(*_Ptup);
-    _Ref       = _STD current_exception();
-    return 0;
-    _CATCH_END
+#ifdef _M_CEE
+#define _WINDOWS_API              __stdcall
+#define _RENAME_WINDOWS_API(_Api) _Api##_clr
+#else // ^^^ _M_CEE // !_M_CEE vvv
+#define _WINDOWS_API              __declspec(dllimport) __stdcall

```

Comment:
Should we mark these functions as `__declspec(dllimport)`, or nothing? I am *far* from an expert here, but we overhauled the STL somewhat recently to avoid mentioning `__declspec(dllimport)` for any functions (it is still necessary for classes/data).

Reply:
We avoided __declspec(dllimport) for functions declared in our own libraries, because Windows folks wanted to be able to build a static lib that didn't care about the CRT build model, if I understand correctly. These functions always must come from a DLL (since they're part of the OS), so using dllimport is safe regardless of CRT build model.

Reply:
Ok.

Reply:
Is it impossible that STL is used to build the same DLL that export those functions?

Reply:
Windows uses the STL, but due to circular dependencies, I don't believe they use it to implement Windows API functions like this.